analyses/cell-type-ewings/scripts/aucell-ews-signatures/01-aucell.R

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env Rscript
+
+# This script is used to run `AUCell` on a single SCE object for a set of marker gene sets 
+# gene sets used are custom gene sets and a set of Ewing specific gene sets from MsigDB
+# the results are exported as a single TSV file with the following columns: 
+# `gene_set`, `barcodes`, `auc`, and `auc_threshold`
+
+project_root <- here::here()
+
+library(optparse)
+
+option_list <- list(
+  make_option(
+    opt_str = c("--sce_file"),
+    type = "character",
+    help = "Path to RDS file containing a processed SingleCellExperiment object from scpca-nf
+      to be used to run AUCell."
+  ),
+  make_option(
+    opt_str = c("--custom_geneset_dir"),
+    type = "character",
+    default = file.path(project_root, "references", "gene_signatures"), 
+    help = "Path to directory containing custom gene sets to use with AUCell. 
+      All TSV files in the directory will be used and must contain the `ensembl_gene_id` column.
+      File names will be used as the name of the gene set."
+  ),
+  make_option(
+    opt_str = c("--msigdb_genesets"),
+    type = "character", 
+    default = file.path(project_root, "references", "msigdb-gene-sets.tsv"),
+    help = "Path to TSV file containing all gene sets from MSigDB to use with AUCell.
+      Must contain columns with `name`, `geneset`, `category`, and `subcategory`."
+  ), 
+  make_option(
+    opt_str = c("--max_rank_threshold"),
+    type = "double",
+    default = 0.01,
+    help = "Percentage of genes detected to set as the `aucMaxRank`. 
+      Must be a number between 0 and 1."
+  ),
+  make_option(
+    opt_str = c("--output_file"),
+    type = "character",
+    help = "Path to file where results will be saved"
+  ),
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 4,
+    help = "Number of multiprocessing threads to use."
+  ),
+  make_option(
+    opt_str = c("--seed"),
+    type = "integer",
+    default = 2025,
+    help = "A random seed for reproducibility."
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# make sure input files exist
+stopifnot(
+  "sce file does not exist" = file.exists(opt$sce_file),
```

Comment:
This will fail with `Error in file.exists(opt$sce_file) : invalid 'file' argument` if an SCE file is not provided in the first place, so you might want to first have a separate check that it's not null.